vertical-pod-autoscaler/updater/priority/update_priority_calculator.go

```diff
@@ -85,44 +101,62 @@ func (calc *UpdatePriorityCalculator) GetSortedPods() []*apiv1.Pod {
 	return result
 }
 
-func (calc *UpdatePriorityCalculator) getUpdatePriority(pod *apiv1.Pod, recommendation *vpa_types.RecommendedPodResources) float64 {
-	var priority float64
+func (calc *UpdatePriorityCalculator) getUpdatePriority(pod *apiv1.Pod, recommendation *vpa_types.RecommendedPodResources) podPriority {
+	outsideRecommendedRange := false
+	scaleUp := false
+	// Sum of requests over all containers, per resource type.
+	totalRequestPerResource := make(map[apiv1.ResourceName]int64)
+	// Sum of recommendations over all containers, per resource type.
+	totalRecommendedPerResource := make(map[apiv1.ResourceName]int64)
 
 	for _, podContainer := range pod.Spec.Containers {
 		recommendedRequest, err := vpa_api_util.GetCappedRecommendationForContainer(podContainer, recommendation, calc.resourcesPolicy)
 		if err != nil {
 			glog.V(2).Infof("no recommendation for container %v in pod %v", podContainer.Name, pod.Name)
 			continue
 		}
-		for resourceName, recommended := range recommendedRequest {
-			var requested *resource.Quantity
-
-			if request, ok := podContainer.Resources.Requests[resourceName]; ok {
-				requested = &request
+		for resourceName, recommended := range recommendedRequest.Target {
+			totalRecommendedPerResource[resourceName] += recommended.MilliValue()
+			minRecommneded, hasMin := recommendedRequest.MinRecommended[resourceName]
+			maxRecommneded, hasMax := recommendedRequest.MaxRecommended[resourceName]
+			if request, hasRequest := podContainer.Resources.Requests[resourceName]; hasRequest {
+				totalRequestPerResource[resourceName] += request.MilliValue()
+				if recommended.MilliValue() > request.MilliValue() {
+					scaleUp = true
+				}
+				if (hasMin && request.MilliValue() < minRecommneded.MilliValue()) ||
+					(hasMax && request.MilliValue() > maxRecommneded.MilliValue()) {
+					outsideRecommendedRange = true
+				}
+			} else {
+				scaleUp = true
+				outsideRecommendedRange = true
```

Comment:
AFAIU this code if we start new pod without request it will be evicted immediately just to assign request. Seems dangerous - I would prefer to let such pod live until we get better confidence in recommendation or it dies by itself (but giving us some usage samples before that)